java/core/src/java/org/apache/orc/impl/ColumnStatisticsImpl.java

```diff
@@ -608,8 +609,23 @@ public void merge(ColumnStatisticsImpl other) {
       OrcProto.StringStatistics.Builder str =
         OrcProto.StringStatistics.newBuilder();
       if (getNumberOfValues() != 0) {
-        str.setMinimum(getMinimum());
-        str.setMaximum(getMaximum());
+        String value = getMinimum();
+        if (value != null && value.length() > 1024) {
```

Comment:
I suggest you make it a constant somewhere and reference that.

I would also consider making it configurable with default value of `1024`.